thunder/__init__.py

```diff
@@ -735,13 +797,16 @@ def fn_(*args, **kwargs) -> Any:
                     NotImplementedError,
                 )
 
+        _nvtx_push("computation_fn")
         result = cache_entry.computation_fn(*inps)
+        _nvtx_pop()
 
         if cache_entry.backward_fn:
             # Run the compiled forward function
             data_for_autograd, (saved_tensors, saved_other) = result
 
             # Connect produced tensors with PyTorch's autograd graph
+            _nvtx_push("connect autograd")
```

Comment:
A Pythonic way to push and pop is using a context manager. Then any exception in the middle wouldn't mess up the annotations.
```py
with nvtx.annotate(message="connect autograd"):
    pass
```

Reply:
Closing this thread as this is tracked in the thread above.